### PR TITLE
[NEW] Add support for hardcoded server URL

### LINKED ISSUE #2177: [IMPROVEMENT] Add support for hardcoded server URL
The IOS client supports RC_SERVER_URL in info.plist file, which allows the user to skip the initial screens that are about selecting a server. It would be great if the Android client supported a similar mechanism and, if set, would take the user directly to the login screen.

### PR DESCRIPTION
**Pull request by cardoso**
@RocketChat/android

Closes #2177 

#### Changes: 

- Adds RC_SERVER_URL to build configs
- Connects automatically to server set in build configs
- Disables chang server/add server modal if RC_SERVER_URL is set

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -62,6 +62,7 @@ android {
 
     buildTypes {
         release {
+            buildConfigField "String", "RC_SERVER_URL", '""'
```
Comment (philipbrito) [inline app/build.gradle]: Move to `defaults.xml` file please. 

string tag can be just `server_url` there.

File: app/build.gradle [hunk 2]
```diff
@@ -70,6 +71,7 @@ android {
         }
 
         debug {
+            buildConfigField "String", "RC_SERVER_URL", '""'
```
Comment (philipbrito) [inline app/build.gradle]: After moving you don't need even to declare it here.

File: app/src/main/java/chat/rocket/android/authentication/server/ui/ServerFragment.kt [hunk 3]
```diff
@@ -86,6 +86,11 @@ class ServerFragment : Fragment(), ServerView {
         deepLinkInfo?.let {
             it.url.toUri().host?.let { host -> text_server_url.hintContent = host }
             presenter.deepLink(it)
+        }.ifNull {
+            if (BuildConfig.RC_SERVER_URL != "") {
```
Comment (philipbrito) [inline app/src/main/java/chat/rocket/android/authentication/server/ui/ServerFragment.kt]: Checking if a string is not empty? Use `.isNotEmpty()`.

File: app/src/main/java/chat/rocket/android/authentication/ui/AuthenticationActivity.kt [hunk 4]
```diff
@@ -74,8 +75,10 @@ class AuthenticationActivity : AppCompatActivity(), HasSupportFragmentInjector {
             presenter.loadCredentials(newServer) { isAuthenticated ->
                 if (isAuthenticated) {
                     showChatList()
-                } else {
+                } else if (BuildConfig.RC_SERVER_URL == "") {
```
Comment (philipbrito) [inline app/src/main/java/chat/rocket/android/authentication/ui/AuthenticationActivity.kt]: Checking if a string is not empty? Use `.isNotEmpty()`.

File: app/src/main/java/chat/rocket/android/chatrooms/ui/ChatRoomsFragment.kt [hunk 5]
```diff
@@ -272,11 +273,13 @@ class ChatRoomsFragment : Fragment(), ChatRoomsView {
     }
 
     private fun setupListeners() {
-        text_server_name.setOnClickListener {
-            ServersBottomSheetFragment().show(
-                activity?.supportFragmentManager,
-                chat.rocket.android.servers.ui.TAG
-            )
+        if (BuildConfig.RC_SERVER_URL == "") {
```
Comment (philipbrito) [inline app/src/main/java/chat/rocket/android/chatrooms/ui/ChatRoomsFragment.kt]: Checking if a string is not empty? Use `.isNotEmpty()`.

File: app/src/main/java/chat/rocket/android/chatrooms/ui/ChatRoomsFragment.kt [hunk 6]
```diff
@@ -272,11 +273,13 @@ class ChatRoomsFragment : Fragment(), ChatRoomsView {
...
+            text_server_name.setOnClickListener {
+                ServersBottomSheetFragment().show(
+                        activity?.supportFragmentManager,
```
Comment (philipbrito) [inline app/src/main/java/chat/rocket/android/chatrooms/ui/ChatRoomsFragment.kt]: Wrong indentation. Please make sure to follow our coding style.

**Review by philipbrito — CHANGES_REQUESTED** [review]

**Comment by cardoso** [comment]
Thanks for your review @filipedelimabrito . I've updated the code accordingly.

File: app/src/main/java/chat/rocket/android/chatrooms/ui/ChatRoomsFragment.kt [hunk 7]
```diff
@@ -272,11 +272,13 @@ class ChatRoomsFragment : Fragment(), ChatRoomsView {
...
     private fun setupListeners() {
-        text_server_name.setOnClickListener {
-            ServersBottomSheetFragment().show(
-                activity?.supportFragmentManager,
-                chat.rocket.android.servers.ui.TAG
-            )
+        if (getString(R.string.server_url).isEmpty()) {
+            text_server_name.setOnClickListener {
+                ServersBottomSheetFragment().show(
+                    activity?.supportFragmentManager,
```
Comment (philipbrito) [inline app/src/main/java/chat/rocket/android/chatrooms/ui/ChatRoomsFragment.kt:278]: Thank you very much for sending this PR @cardoso! I'm glad to see you here.

Just a tip: `chat.rocket.android.servers.ui.TAG` fits on the same line where `activity?.supportFragmentManager` is declared here - respecting the coding style. But it's fine!

Thank you! 👊

**Review by philipbrito — APPROVED** [review]

**Pull request merged**